clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -907,7 +907,8 @@ void CodeGenModule::Release() {
   if (Context.getTargetInfo().getTriple().isWasm())
     EmitMainVoidAlias();
 
-  if (getTriple().isAMDGPU()) {
+  if (getTriple().isAMDGPU() ||
+      (getTriple().isSPIRV() && getTriple().getVendor() == llvm::Triple::AMD)) {
```

Comment:
I'm wondering if we should add `isAMD` to `llvm::Triple` or something.

Reply:
I didn't know the vendor got used for anything.

Reply:
> I'm wondering if we should add `isAMD` to `llvm::Triple` or something.

We do have `isAMDGCN`, but that reflects the Arch. I guess it might be convenient sugar to have a predicate on vendors as well? I don't find the manual check excessively offensive FWIW.

Reply:
> I didn't know the vendor got used for anything.

This matches how we've documented it when we added the AMDGCN flavoured SPIR-V, and seemed to reflect the idea that this is SPIRV(64) with customisations for AMD as the vendor; do you think it is or can become problematic?